### ISSUE TITLE
contribution key and GPL compatability

### ISSUE THREAD

**Issue by yorambh** [Wunderfitz/harbour-piepmatz#44]
Hello.
just to make clear: I did purchase contribution key because I support this application and free-software in general. however:
according to: https://www.gnu.org/licenses/gpl-faq.en.html#DoesTheGPLAllowRequireFee
it seem like you can charge for getting a copy from you, but you can not charge for using an existing copy, which means, locking the application until one purchase a key, may be in compatible wit GPL.

please verify this and make it clear.

**Comment by Wunderfitz** [comment]
Hi @yorambh,

first thanks for purchasing a key, really appreciated! Also thanks for your question, let me outline why I think that the pre-packaged version and the source code are perfectly compatible with the GPL(v3):

The contribution key management system actually doesn't require anybody who receives the software to pay a fee or notify me as the developer. The binary (and of course the source code) is freely distributable and installable on any device where you want to do it. You can compile and execute it wherever you want without notifying anybody or paying anything. The contribution system only comes into play in the version which I compiled and packaged. So in the end this fits to the condition of the FAQ which you've mentioned: "You can charge people a fee to get a copy from you."

The next sentence "You can't require people to pay you when they get a copy from someone else." is also perfectly fulfilled: Anybody who wants to create a copy of Piepmatz without the contribution management system can grab the source code, remove the things which they don't want, compile, package and ship it. Of course, that would severely limit the sales, but that's part of the deal. I used a lot of free software and I also give a lot back. I only ask my users to give something back as well - either in money or in code/translation contributions. As a simple donation system doesn't work out as several examples in the industry prove, I tried to apply a different model. 

I'm really curious if this works out because if that scales and people pay for free software if they don't contribute in other ways, it could boost free software even more as developers can get money out out it instead of going closed source or using advertisements/seeling your personal data. If somebody who doesn't contribute anything to the features strips the contribution system and heavily distributes it just to avoid paying the value of a beer to a developer who spent days and weeks on good software - well, that's their free choice. However, that might lead to even less developers in the OSS world, especially for consumer products and that would make me very sad.

To sum it up: None of the four essential freedoms which are mentioned in https://www.gnu.org/philosophy/free-sw.html.en are violated by me in any way. It's quite the opposite. It's an attempt to bring Open Source Software together with a commercial sales model for consumers so that developers are encouraged to publish their code for the benefit of everybody by earning some money for their hard work.

I hope, I could answer your question. In case that you have any other questions, feel free to reopen this issue.

Best regards, Sebastian

**Comment by yorambh** [comment]
Hello and thanks for the answer
after doing some reading, I am convinced that GPL permits any code, including code that limits or disables functionality
your interpretation, therefore, seems correct
Here is a blog post that explains it simply: https://davefx.com/en/2017/05/gpl-software-requiring-payments-to-work/

here is a key quote from FSF answer to similar concern:
  Hello and thank you for writing in.

> The GPL doesn’t prohibit locks or schemes such as these, as long as the recipient of the software can modify or remove them as per the terms of the GPL. It isn’t the lock itself that is prohibited; it’s restricting others from studying, modifying, or removing that lock that the GPL prohibits. It would also be a violation of the GPL to add licensing terms which prohibit the recipient of the software from removing such feature-lock schemes.
>
> You may also find the following interesting: http://www.gnu.org/licenses/gpl-faq.html#DRMProhibited 

Thanks
Yoram Bar-Haim

**Comment by Wunderfitz** [comment]
Thanks, that blog post is really a good read!
